### PR TITLE
Port domain tests to airgun

### PR DESCRIPTION
**Pull request by bsquizz**
Depends on https://github.com/SatelliteQE/airgun/pull/87

### CODE REVIEW

File: tests/foreman/ui_airgun/test_domain.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    """
+    Create a domain and optionally assert that it is created
+    """
+    if not name:
+        name = DOMAIN % gen_string('alpha')
+    session.domain.create({
+        'domain.dns_domain': name,
+        'domain.full_name': name,
+    })
+    if assert_created:
```
Comment (lpramuk) [inline tests/foreman/ui_airgun/test_domain.py]: +1

**Comment by bsquizz** [comment]
Chatted with @abalakh -- I'll leave the ported tier1's in ui_airgun since they're working, but take the `@tier1` decorator out so they are not run in CI. I moved the tier1's back into ui. And I removed the _create_domain helper method. Personally I think it's a good idea consolidate code like this using fixtures/helper methods. If we want to change the logic for how we create a domain in this test in the future it would be a lot less work to update 1 method :) But anyways... changed it back for now for this PR.

File: tests/foreman/ui_airgun/test_domain.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    with session:
+        with pytest.raises(AssertionError) as context:
+            session.domain.create({
+                'domain.dns_domain': name,
+                'domain.full_name': name,
+            })
+        assert 'errors present' in str(context.value)
+
+
+@run_only_on('sat')
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_domain.py]: remove these decorators from all tests as that is just very old legacy code

File: tests/foreman/ui_airgun/test_domain.py [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+            }
+        ]
+    }
+    with session:
+        name = DOMAIN % gen_string('alphanumeric')
+        session.domain.create({
+            'domain.dns_domain': name,
+            'domain.full_name': name,
+        })
+        session.domain.update(name, update_values)
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_domain.py]: we making our ported tests much more robust, so please read back domain values and validate that values are really updated

File: tests/foreman/ui_airgun/test_domain.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_set_parameter_long(session):
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_domain.py]: not sure that previous test case is needed. We can make one with previous and verify only '255' values

File: tests/foreman/ui_airgun/test_domain.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+            }
+        ]
+    }
+    with session:
+        name = DOMAIN % gen_string('alphanumeric')
+        session.domain.create({
+            'domain.dns_domain': name,
+            'domain.full_name': name,
+        })
+        session.domain.update(name, update_values)
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_domain.py]: validate update here too

File: tests/foreman/ui_airgun/test_domain.py (hunk 3 above)
Comment (bsquizz) [reply tests/foreman/ui_airgun/test_domain.py]: done, now reading the view and asserting the param values.

**Comment by bsquizz** [comment]
Validating expected values after `session.domain.update`, removed `@run_only_on`, combined the 2 create_domain tests into 1 test that uses long name, combined the parameter tests into a parametrized test (setting param value w/ 255 chars, and setting param w/ blank value)

**Comment by oshtaier** [comment]
I see that
```
+    def fill(self, values):
+        """Fill parameter entries. Existing values will be overwritten.
+
+        Updates name/value pairs if the name is already in the list.
+
+        If you desire to intentionally add a duplicate value, use self.add()
```
not sure, but maybe we can make logic not so smart and let it do directly what we ask to do

File: tests/foreman/ui_airgun/test_domain.py [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    parametrize,
+    tier2,
+    upgrade
+)
+
+
+def valid_domain_names(length):
+    """Returns pytest parametrize kwargs for valid domain names."""
+    max_len = 255 - len(DOMAIN % '')
+    if max_len - length < 0:
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_domain.py]: Nitpick: maybe `if length > max_len`? :)

File: tests/foreman/ui_airgun/test_domain.py [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+from robottelo.constants import DOMAIN
+from robottelo.datafactory import parametrized
+from robottelo.decorators import (
+    parametrize,
+    tier2,
+    upgrade
+)
+
+
+def valid_domain_names(length):
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_domain.py]: Please move this and the next func to robottelo/datafactory.py (take a look at valid_data_list for the reference).
Also i'd make `length` optional to make it consistent with other datafactory generators - if length is not passed then use random one.

File: tests/foreman/ui_airgun/test_domain.py (hunk 6 above)
Comment (bsquizz) [reply tests/foreman/ui_airgun/test_domain.py]: yes, that's much better ;)

**Comment by bsquizz** [comment]
```
(robottelo_venv)  bsquizza@squizzlap  ~/repos/robottelo   port_domain  pytest tests/foreman/ui_airgun/test_domain.py                            
====================================================================================================== test session starts ======================================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/bsquizza/repos/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 20 items                                                                                                                                                                                                              
2018-06-14 14:54:14 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_domain.py ....................                                                                                                                                                               [100%]

================================================================================================== 20 passed in 765.79 seconds ==================================================================================================
(robottelo_venv)  bsquizza@squizzlap  ~/repos/robottelo   port_domain  
```

**Pull request merged**